autoinstall-schema.json

```diff
@@ -461,6 +461,18 @@
                 "additionalProperties": false
             }
         },
+        "active_directory": {
```

Comment:
Most autoinstall things are using dash instead of underscore.  `search_drivers` and `preserve_sources_list` are  exceptions but I want to fix them now.  In the interest on standardizing autoinstall, I'm going to ask you to use a dash.

Reply:
Thanks - but examples/autoinstall-ad.yaml needs the tweak, and tests passed?  Are the packages being installed unconditionally?

Reply:
Well observed. There is a logic error in the test. I'm finishing running the integration tests locally to confirm the fix.

Reply:
Commit 121cf3a18ac21503e9aa5d12e54deb364773852f reveals the logic error and testing against it would crash. The following aeac073dca0d1998ec1b394fc17a0b3a5e8ef8ea fixes the autoinstall file and should not crash the integration test.

Reply:
Should we consider exposing the `variant` for the `autoinstall`?

Reply:
I'm not sure what problem exposing the variant would solve.  If this is required in autoinstall then we have made a critical error.

If someone wants to configure active-directory, then they should do so.  If not, the omit the section.

Reply:
Currently "active_directory" is only available for the desktop variant (because only for that variant we add the model as part of the POST_INSTALL_NAMES set, what causes the extra packages to be installed in the target).

But you're right. That condition is an implementation detail that will be sorted out when the server variant also becomes interested in joining an Active Directory domain.